### PR TITLE
[Fix](inverted index) need_read_data only supports DUP_KEYS

### LINKED ISSUE #23566: [Bug] bitmap index will get wrong count(*) result after execute delete sql
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/doris/issues?q=is%3Aissue) and found no similar issues.

### Version

version: 2.0.1-rc03

### What's Wrong?

When I delete some record from the table which has bitmap index, then I execute sql ```select count(*) from xxx where xxx```, it will get wrong result

### What You Expected?

It return right result

### How to Reproduce?

-- 建表
CREATE TABLE `test2` (
  `create_time` datetime NOT NULL COMMENT '',
  `vid` varchar(64) NOT NULL COMMENT '',
  `report_time` datetime NULL COMMENT '',
  `block_version` int(11) NULL COMMENT '',
  `vehicle_mode` int(11) NULL COMMENT '',
  `usage_mode` int(11) NULL COMMENT '',
INDEX vid_bitmap_index (`vid`) USING BITMAP COMMENT '为vid添加位图索引'
) ENGINE=OLAP
UNIQUE KEY(`create_time`, `vid`, `report_time`)
COMMENT 'OLAP'
PARTITION BY RANGE(`create_time`)
(PARTITION p20230820 VALUES [('2023-08-20 00:00:00'), ('2023-08-21 00:00:00')),
PARTITION p20230821 VALUES [('2023-08-21 00:00:00'), ('2023-08-22 00:00:00')),
PARTITION p20230822 VALUES [('2023-08-22 00:00:00'), ('2023-08-23 00:00:00')),
PARTITION p20230823 VALUES [('2023-08-23 00:00:00'), ('2023-08-24 00:00:00')),
PARTITION p20230824 VALUES [('2023-08-24 00:00:00'), ('2023-08-25 00:00:00')),
PARTITION p20230825 VALUES [('2023-08-25 00:00:00'), ('2023-08-26 00:00:00')),
PARTITION p20230826 VALUES [('2023-08-26 00:00:00'), ('2023-08-27 00:00:00')),
PARTITION p20230827 VALUES [('2023-08-27 00:00:00'), ('2023-08-28 00:00:00')),
PARTITION p20230828 VALUES [('2023-08-28 00:00:00'), ('2023-08-29 00:00:00')),
PARTITION p20230829 VALUES [('2023-08-29 00:00:00'), ('2023-08-30 00:00:00')))
DISTRIBUTED BY HASH(`vid`) BUCKETS AUTO
PROPERTIES (
"replication_allocation" = "tag.location.default: 1",
"is_being_synced" = "false",
"dynamic_partition.enable" = "true",
"dynamic_partition.time_unit" = "DAY",
"dynamic_partition.time_zone" = "Asia/Shanghai",
"dynamic_partition.start" = "-30",
"dynamic_partition.end" = "1",
"dynamic_partition.prefix" = "p",
"dynamic_partition.replication_allocation" = "tag.location.default: 1",
"dynamic_partition.buckets" = "10",
"dynamic_partition.create_history_partition" = "true",
"dynamic_partition.history_partition_num" = "-1",
"dynamic_partition.hot_partition_num" = "0",
"dynamic_partition.reserved_history_periods" = "NULL",
"dynamic_partition.storage_policy" = "",
"storage_format" = "V2",
"light_schema_change" = "true",
"disable_auto_compaction" = "false",
"enable_single_replica_compaction" = "false"
);

-- 插入3条数据
insert into test2(create_time,vid,report_time,block_version,vehicle_mode,usage_mode) values('2023-08-25 10:00:00','123','2023-08-25 10:00:00',1,1,1);
insert into test2(create_time,vid,report_time,block_version,vehicle_mode,usage_mode) values('2023-08-25 11:00:00','123','2023-08-25 11:00:00',2,2,2);
insert into test2(create_time,vid,report_time,block_version,vehicle_mode,usage_mode) values('2023-08-25 12:00:00','123','2023-08-25 12:00:00',3,3,3);

-- 删出1条
delete from test2 where vid="123" and report_time="2023-08-25 12:00:00" and create_time="2023-08-25 12:00:00";

--执行查询sql
-- sql1
select count(\*) from test2;  -- result is 2
-- sql2
select count(\*) from test2 where vid='123';  -- but this result is 3, exepected 2
-- sql3
select count(*) from test2 where create_time>='2023-08-25 10:00:00'; -- result is 2
-- sql4
select count(CASE when vid='123' then 1 else null end) from test2 where vid='123'; -- reuslt is 2
--sql5
select * from test2 where vid='123'; -- result has 2 records

In above sqls, the second sql expected result is 2, but it return 3;

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by airborne12**
## Proposed changes

Issue Number: close #23566 

need_read_data optimization does not work properly when table is UNQIUE_KEY MOR

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by airborne12** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp [hunk 1]
```diff
@@ -938,6 +938,10 @@ Status SegmentIterator::_apply_inverted_index_on_block_column_predicate(
 }
 
 bool SegmentIterator::_need_read_data(ColumnId cid) {
+    // for safety reason, only support DUP_KEYS
+    if (_opts.tablet_schema->keys_type() != KeysType::DUP_KEYS) {
```
Comment (xiaokang) [inline be/src/olap/rowset/segment_v2/segment_iterator.cpp:942]: consider unique table with MOW

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 47.67 seconds
 stream load tsv:          532 seconds loaded 74807831229 Bytes, about 134 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          64 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          31 seconds loaded 861443392 Bytes, about 26 MB/s
 insert into select:          28.7 seconds inserted 10000000 Rows, about 348K ops/s
 storage size: 17162225136 Bytes

File: regression-test/suites/index_p0/test_bitmap_index.groovy [hunk 2]
```diff
@@ -269,4 +269,102 @@ suite("test_bitmap_index") {
...
+        sql "insert into ${tbName4}(create_time,vid,report_time,block_version,vehicle_mode,usage_mode) values('2023-08-25 11:00:00','123','2023-08-25 11:00:00',2,2,2);"
+        sql "insert into ${tbName4}(create_time,vid,report_time,block_version,vehicle_mode,usage_mode) values('2023-08-25 12:00:00','123','2023-08-25 12:00:00',3,3,3);"
+        qt_sql "desc ${tbName4};"
+        qt_sql "SHOW INDEX FROM ${tbName4};"
+        sql "delete from ${tbName4} where vid='123' and report_time='2023-08-25 12:00:00' and create_time='2023-08-25 12:00:00';"
+        qt_sql "select count() from ${tbName4}; "
+        qt_sql "select count() from ${tbName4} where vid='123'; "
+        qt_sql "select count(*) from ${tbName4} where create_time>='2023-08-25 10:00:00';"
+        qt_sql "select count(CASE when vid='123' then 1 else null end) from ${tbName4} where vid='123';"
+        qt_sql "select * from ${tbName4} where vid='123';"
```
Comment (airborne12) [inline regression-test/suites/index_p0/test_bitmap_index.groovy:351]: need to order by

**Review by htyoung — COMMENTED** [review]
qt_sql "select count() from ${tbName4}; "
qt_sql "select count() from ${tbName4} where vid='123'; "

建议更改成

qt_sql "select count(\*) from ${tbName4}; "
qt_sql "select count(\*) from ${tbName4} where vid='123'; "

原issue中的count(\*)被转义了

**Comment by airborne12** [comment]
> qt_sql "select count() from ${tbName4}; " qt_sql "select count() from ${tbName4} where vid='123'; "
> 
> 建议更改成
> 
> qt_sql "select count(*) from ${tbName4}; " qt_sql "select count(*) from ${tbName4} where vid='123'; "
> 
> 原issue中的count(*)被转义了

count() is the same with count(*) in doris

File: be/src/olap/rowset/segment_v2/segment_iterator.cpp (hunk 1 above)
Comment (airborne12) [reply be/src/olap/rowset/segment_v2/segment_iterator.cpp:950]: not support MoW now, because MoW is not always generate delete bitmap for delete predicate

**Comment by airborne12** [comment]
run buildall

**Review by qidaye — APPROVED** [review]
LGTM

**Comment by airborne12** [comment]
run buildall

**Comment by airborne12** [comment]
run buildall

**Review by eldenmoon — APPROVED** [review]
LGTM

**Pull request merged**